tests/integration/schema/similarity_test.go

```diff
@@ -0,0 +1,322 @@
+// Copyright 2025 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package schema
+
+import (
+	"testing"
+
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+)
+
+func TestSchemaInstrospection_SimilarityCapableFieldIntArray(t *testing.T) {
+	test := testUtils.TestCase{
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Users {
+						someVector: [Int!]
+					}
+				`,
+			},
+			testUtils.IntrospectionRequest{
+				Request: `
+					query {
+						__type (name: "Users") {
+							name
+							fields {
+								name
+								args {
+									name
+									type {
+										name
+										inputFields {
+											name
+											type {
+												name
+												kind
+												ofType {
+													name
+													kind
+													ofType {
+														name
+														kind
+														ofType {
+															name
+															kind
+														}
+													}
+												}
+											}
+										}
+									}
+								}
+							}
+						}
+					}
+				`,
+				ContainsData: map[string]any{
+					"__type": map[string]any{
+						"name": "Users",
+						"fields": []any{
+							map[string]any{
+								"name": "_similarity",
+								"args": []any{
+									map[string]any{
+										"name": "someVector",
+										"type": map[string]any{
+											"inputFields": []any{
+												map[string]any{
+													"name": "vector",
+													"type": map[string]any{
+														"kind": "NON_NULL",
+														"name": any(nil),
+														"ofType": map[string]any{
+															"kind": "LIST",
+															"name": any(nil),
```

Comment:
suggestion: `"name": nil,` should work fine

Reply:
Indeed. Changed.